File: solutions/csharp/0345-reverse-vowels-of-a-string.cs
public class Solution {
    public string ReverseVowels(string s) {
        int start = 0, end = s.Length - 1;
        HashSet<char> vowels = new HashSet<char> {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        char[] arr = s.ToCharArray();

        while (start < end) {
            if (vowels.Contains(arr[start]) && vowels.Contains(arr[end])) {
                char temp = arr[start];
                arr[start++] = arr[end];
                arr[end--] = temp;
            } else if (vowels.Contains(arr[start])) {
                end--;
            } else {
                start++;
            }
        }
        return new string(arr);
    }
}

File: solutions/java/0345-reverse-vowels-of-a-string.java
public class Solution {
    public String reverseVowels(String s) {
        int start = 0, end = s.length() - 1;
        Set<Character> vowels = new HashSet<>(Arrays.asList('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'));
        char[] arr = s.toCharArray();

        while (start < end) {
            if (vowels.contains(arr[start]) && vowels.contains(arr[end])) {
                char temp = arr[start];
                arr[start++] = arr[end];
                arr[end--] = temp;
            } else if (vowels.contains(arr[start])) {
                end--;
            } else {
                start++;
            }
        }
        return new String(arr);
    }
}
